wallet-core/src/ffi.rs

```diff
@@ -166,3 +137,38 @@ pub unsafe fn balance(
 
     ErrorCode::Ok
 }
+
+/// Pick the notes to be used in a transaction from an owned notes list.
+#[no_mangle]
+pub unsafe fn pick_notes(
+    seed: &Seed,
+    index: u8,
+    value: *const u64,
+    notes_ptr: *mut u8,
+) -> ErrorCode {
+    let vk = derive_phoenix_vk(seed, index);
+
+    let notes: owned::NoteList = mem::from_buffer(notes_ptr)?;
+
+    let notes = pick::notes(&vk, notes, *value);
+
+    let bytes = to_bytes::<_, NOTES_BUFFER_SIZE>(&notes)
+        .or(Err(ErrorCode::ArchivingError))?;
+
+    let len = bytes.len().to_le_bytes();
+
+    ptr::copy_nonoverlapping(len.as_ptr(), notes_ptr, 4);
+    ptr::copy_nonoverlapping(bytes.as_ptr(), notes_ptr.add(4), bytes.len());
+
+    ErrorCode::Ok
+}
+
+/// Gets the bookmark from the given note.
+#[no_mangle]
+pub unsafe fn bookmark(leaf_ptr: *const u8, bookmark: *mut u64) -> ErrorCode {
```

Comment:
Why not keep the name `position`?

Reply:
Because `position` refers to an internal implementation detail about how we store the note, it's acceptable to use it at the low-level API. However, when we abstract such information, `bookmark` is more appropriate. When a client wants to `sync` (read data) to a specific network (protocol) or a specific contract, if having a block height is not sufficient (or for performance reasons it's better to implement something custom), the `bookmark` is the type that tells you where to continue syncing next time.

It's like reading a book: you set a bookmark to remember where to continue the next time you read.

A position is specific to our internal implementation and leaks implementation details, whereas a bookmark (a generic 8-byte ID) does not.

Also, since we didn't cancel the implementation of drivers but just postponed it, it's good for w3sper to deal with just one type (`Bookmark`) for both protocol drivers and third-party contracts.

Additionally, remember that this naming is only for the FFI, where the low-level API connects to the high-level ones and where we start to abstract details.